[15.0][MIG]l10n_es_pos

manuelregidor opened the pull request:
Standard migration

Commit:
[ADD] Add l10n_es_pos

sbidoul committed:
[UPD] prefix versions with 8.0

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

xparsox committed:
[MIG] l10n_es_pos

acysos committed:
[ADD] l10n_es_pos: posbox support and translations

chienandalu committed:
[MIG] l10n_es_pos: Migration to 11.0

jonerikceberio committed:
[FIX] l10n_es_pos: typo in oldname (#886)

jonerikceberio committed:
[IMP] l10n_es_pos: Handle the sequence properly (#902)

pedrobaeza committed:
[FIX] l10n_es_pos: Import uninstall_hook

chienandalu committed:
[IMP] l10n_es_pos: move simpl. inv. reference to standard field (#945)

This way we don't need to make glue modules over l10n_es_pos every time
we need to make use of that info for a generic purpose.

chienandalu committed:
[MIG] l10n_es_pos: Migration to 12.0

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (31 of 31 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_pos/es/

chienandalu committed:
[FIX] l10n_es_pos: XmlReceipt doesn't show customer

chienandalu committed:
[FIX] l10n_es_pos: prevent duplicate or missing orders

chienandalu committed:
[FIX] l10n_es_pos: errors due to context

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (32 of 32 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_pos/es/

CarlesAntoli committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (32 of 32 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_pos/ca/

CarlosRoca13 committed:
[IMP] l10n_es_pos: black, isort, prettier

CarlosRoca13 committed:
[MIG] l10n_es_pos: Migration to v13.0

JordiBForgeFlow committed:
rename category of l10n_es_pos, as 'Point of Sale' is not valid in 13.0

danimv5 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (25 of 25 strings)

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_pos/es/

danimv5 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (25 of 25 strings)

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_pos/ca/

joao-p-marques committed:
[IMP] l10n_es_pos: black, isort, prettier

joao-p-marques committed:
[MIG] l10n_es_pos: Migration to 14.0

TT26435

joao-p-marques committed:
[IMP] l10n_es_pos: Export ticket correctly

Co-authored-by: Enrique Martín Zabalza <enriquemartin@digital5.es>

oca-travis committed:
[UPD] Update l10n_es_pos.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_pos/

danimv5 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (29 of 29 strings)

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_pos/ca/

rauferdeveloper committed:
[14.0][FIX] l10n_es_pos: Incorrect label texts and translations

oca-travis committed:
[UPD] Update l10n_es_pos.pot

OCA-git-bot committed:
l10n_es_pos 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_pos/

Commit:
[FIX] Refactoring use of l10n_es_unique_id

manuelregidor committed:
[IMP]l10n_es_pos: black, isort, prettier

manuelregidor committed:
[MIG]l10n_es_pos: Migration to 15.0

manuelregidor commented:
@HaraldPanten @ValentinVinagre

ValentinVinagre commented on l10n_es_pos/__manifest__.py:
```diff
@@ -9,9 +9,14 @@
     "website": "https://github.com/OCA/l10n-spain",
     "license": "AGPL-3",
-    "version": "14.0.1.1.0",
+    "version": "15.0.1.1.0",
```
Debería de ser 15.0.1.0.0

HaraldPanten commented:
/ocabot migration l10n_es_pos

HaraldPanten reviewed (approved):
Revisión funcional

AntoniRomera commented:
@HaraldPanten @manuelregidor @ValentinVinagre 
Antes de aprobar este PR, podeis revisar #1909 y consensuar un comportamiento del producto para poder ir avanzando.

HaraldPanten commented:
@manuelregidor puedes añadir https://github.com/OCA/l10n-spain/pull/1909 ?

HaraldPanten commented:
@AntoniRomera @pedrobaeza Está listo para la revisión y añadido el commit correspondiente con el FIX que se añadió para V14.

Saludos.

ValentinVinagre reviewed (approved):
LGTM 👍🏻

HaraldPanten commented:
En principio está OK.

@pedrobaeza lo tiramos adelante?

pedrobaeza commented on l10n_es_pos/models/pos_config.py:
```diff
@@ -78,7 +78,7 @@ def _onchange_l10n_iface_l10n_es_simplified_invoice(self):
         ctx = dict(self._context)
         ctx.update(copy_pos_config=True)
-        return super(PosConfig, self.with_context(ctx)).copy(default)
+        return super(PosConfig, self.with_context(**ctx)).copy(default)
```
No hay necesidad de esto, porque ya se está copiando antes el contexto que había.

manuelregidor replied:
@pedrobaeza Disculpa, pero creo que no acabo de entender a qué te refieres.

pedrobaeza replied:
Decía que no hagas una copia del contexto, si no llamarlo directamente añadiendo el nuevo elemento. ¿O hay alguna razón para copiar el contexto?

pedrobaeza reviewed (commented):
El PoS ha sido rehecho en OWL. Resulta extraño que el código JS anterior funcione sin más. ¿Lo habéis probado?

HaraldPanten commented:
En stand by a la espera de https://github.com/OCA/l10n-spain/issues/2047 para integrar últimos cambios.

Commit:
[IMP] l10n_es_pos:
- En el caso de tener red, el sistema para obtener la numeración de las facturas simplificadas hara una peticion al servidor.
- En caso contrario usaremos el uid generado por odoo.

HaraldPanten commented:
@pedrobaeza Listo para revisar

pedrobaeza commented on l10n_es_pos/models/pos_config.py:
```diff
@@ -76,9 +76,10 @@ def _onchange_l10n_iface_l10n_es_simplified_invoice(self):
-        return super(PosConfig, self.with_context(ctx)).copy(default)
+        return super(
+            PosConfig,
+            self.with_context(**dict(self._context).update(copy_pos_config=True)),
```
Sigue haciendo una copia aquí...

@chienandalu había una razón para hacer una copia?

chienandalu replied:
No creo. Supongo que es código feote del pasado.
```suggestion
            self.with_context(copy_pos_config=True),
```

Mat-moran commented:
Hola buenas, solo añadir un par de pequeños detalles:
- A la hora de buscar tickets antiguos, si se a modificado la secuencia de la factura simplificada por ejemplo metiendo el año (TPV0001 --> TPV20220001), no encuentra los tickets correspondientes.
- No es posible elegir la secuencia durante la configuración del TPV, es posible modificar la que hay seleccionada pero no cambiarla.

manuelregidor commented:
@Mat-moran Gracias por los comentarios. Te comento cada uno de los puntos.

1. ¿Desde dónde realizas la búsqueda, desde el POS o desde el backend? Hay que tener en cuenta que la secuencia de factura simplificada se establece al generar el pedido y, en caso de modificar posteriormente el código de la secuencia que se está utilizando, este valor en cada uno de los pedidos no se modifica.
2. El campo de secuencia de la configuración del POS es de solo lectura, porque se genera automáticamente al crear un nuevo POS. Es decir, que no puede seleccionarse, sino que se asigna una secuencia automáticamente que más tarde puede editarse, pero no cambiarse por otra. Es así como está planteado el módulo y desde mi punto de vista habría que proponer un IMP una vez la migración esté aceptada si es que se decide que el funcionamiento tendría el que describes.

Mat-moran commented:
@manuelregidor gracias por la respuesta,

1. lo busco desde el POS. El problema no es que no modifica la numeración, eso lo hace correctamente, pero no encuentra el pedido con numeraciones que incluyan el año, por ejemplo. El problema debe venir del Regex que limita la búsqueda, que se amplio en este módulo para que permitiera buscar numeraciones acordes con la legislación española después de un cambio que introdujo Odoo y que hacía que no se pudieran buscar la numeraciones de la factura simplificada.
2. Entendido Quizá sería interesante cuando esto este mergeado extender el módulo para que permita cambiarla, ya que si por algún motivo tienes que cambiar la numeración, tendrías que hacer otro pos y con ello perderías todos los análisis de ventas asociados.

Si te parece abro un issue para comentar esto y vemos que se decide finalmente.

manuelregidor commented:
@Mat-moran He aplicado algunos cambios en el módulo. Podrías echar un vistazo? Gracias.

Mat-moran commented:
> @Mat-moran He aplicado algunos cambios en el módulo. Podrías echar un vistazo? Gracias.

Acabo de probar, has introducido la posibilidad de buscar por factura simplificada n?
![image](https://user-images.githubusercontent.com/4810013/176116093-d45903be-cadc-4bdd-9940-0d1d39ce86f9.png)

Busca perfectamente, gracias por incluir estos cambios.

manuelregidor commented:
> > @Mat-moran He aplicado algunos cambios en el módulo. Podrías echar un vistazo? Gracias.
> 
> Acabo de probar, has introducido la posibilidad de buscar por factura simplificada n? ![image](https://user-images.githubusercontent.com/4810013/176116093-d45903be-cadc-4bdd-9940-0d1d39ce86f9.png)
> 
> Busca perfectamente, gracias por incluir estos cambios.

Sí, he añadido esa posibilidad.

manuelregidor commented:
@Mat-moran Después de estos cambios, podrías revisar y aprobar si lo ves todo correcto, por favor? Gracias!

Mat-moran reviewed (commented):
Le he pegado una revisión general parece todo correcto a excepción de lo siguiente:

- Cuando estas en modo `Gestión de inventario = Tiempo real` y te quedas sin conexión a internet, te deja hacer ventas que se quedan pendientes de enviarse a BD cuando vuelva la conexión. En ese estado no asigna numero de factura simplificada  aunque posteriormente cuando vuelve a tener conexión la numeración de la factura simplificada es correcta

![image](https://user-images.githubusercontent.com/4810013/176678771-45b7add6-44bc-49bf-bc48-93d8d75fde97.png)

Esto error no se introduce con la migración esta, ya que también ocurre en versión 14.0, no se que hacer en este caso apruebo? o crees que puede arreglar eso de forma sencilla?

manuelregidor commented:
> Le he pegado una revisión general parece todo correcto a excepción de lo siguiente:
> 
> * Cuando estas en modo `Gestión de inventario = Tiempo real` y te quedas sin conexión a internet, te deja hacer ventas que se quedan pendientes de enviarse a BD cuando vuelva la conexión. En ese estado no asigna numero de factura simplificada  aunque posteriormente cuando vuelve a tener conexión la numeración de la factura simplificada es correcta
> 
> ![image](https://user-images.githubusercontent.com/4810013/176678771-45b7add6-44bc-49bf-bc48-93d8d75fde97.png)
> 
> Esto error no se introduce con la migración esta, ya que también ocurre en versión 14.0, no se que hacer en este caso apruebo? o crees que puede arreglar eso de forma sencilla?

Quizás podríamos aplicarle un FIX posteriormente. Qué opinas?

Mat-moran commented:
Por mi parte perfecto, luego apruebo que ahora no puedo.

pedrobaeza commented:
Incluye en el ROADMAP esa cuestión

manuelregidor commented:
> Incluye en el ROADMAP esa cuestión

Hecho

pedrobaeza commented:
/ocabot merge nobump

pedrobaeza commented:
/ocabot merge nobump
